### PR TITLE
Add tiny model

### LINKED ISSUE #31804: TinyModel addition
### Model description

https://github.com/noanabeshima/tiny_model

It's a small language model trained on TinyStories for interpretability with sparse autoencoders and transcoders added. It has no layernorms (this helps with interpretability) which makes it not fit with any existing model architecture in the transformers library.

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

The implementation is here:
https://github.com/noanabeshima/tiny_model/blob/main/tiny_model/lm.py

The weights are here:
https://huggingface.co/noanabeshima/tiny_model/blob/main/tiny_model.pt

The default config corresponding to the weights is:

        d_model=768,
        n_layers=4,
        n_heads=16,
        max_seq_len=256,
        vocab_size=10_000

I am the author.

### PR DESCRIPTION
**Pull request by sushmanthreddy**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=47782)

This PR adds native Transformers support for [TinyModel](https://github.com/noanabeshima/tinymodel), a small decoder-only language model trained on TinyStories.

  This follows the discussion in #31804, especially [Arthur’s last comment](https://github.com/huggingface/transformers/issues/31804#issuecomment-2395346767). He suggested keeping the implementation minimal,
  following the Transformers API—particularly `from_pretrained`—and avoiding extras such as past-key-value caching when the model does not need them. That is the approach taken here.

  The PR adds:

  - `TinyModelConfig`, `TinyModel`, and `TinyModelForCausalLM`.
  - Native `AutoConfig`, `AutoModel`, and `AutoModelForCausalLM` registration.
  - A small handwritten implementation matching the original architecture:
    - learned absolute positional embeddings;
    - separate bias-free query, key, and value projections;
    - biased attention output projection;
    - ReLU feed-forward layers;
    - no normalization or dropout;
    - untied token embeddings and language-modeling head.
  - Cacheless generation. `use_cache=True` and `past_key_values` are explicitly rejected because the original model does not use KV caching.
  - Standard `attention_mask` and `position_ids` support, including padded inputs.
  - A strict converter for all three released checkpoints:
    - `tiny_model.pt`
    - `tiny_model_2L_1E.pt`
    - `tiny_model_2L_3E.pt`
  - Documentation and model/conversion tests.

  The released checkpoints are plain PyTorch state dictionaries. The converter validates their keys, shapes, layer count, and dtype before writing a standard Transformers checkpoint. The converted model can then be
  loaded normally with `TinyModelForCausalLM.from_pretrained(...)`.

  The original text pipeline uses custom normalization and token-ID remapping assets that are not included with the published checkpoints. Rather than silently using an incompatible tokenizer, this integration
  currently accepts token IDs directly. Tokenizer support can be added later if those assets become available.

  One remaining question for @noanabeshima: I could not find a license file in the pinned implementation repository. Could you confirm the intended license for the source code?

  Fixes #31804

  ## Before submitting

  - [x] This PR fixes a typo or improves the docs.
  - [x] Did you read the contributor guideline and the Pull Request checks?
  - [x] This contribution was discussed in #31804.
  - [x] The documentation was updated.
  - [x] Model and checkpoint-conversion tests were added.

  ## Validation

  - `pytest -q tests/models/tiny_model/`
    - 154 passed
    - 166 skipped
  - `make fix-repo`
    - all 15 checks passed
  - Current PR CI passes, including all eight `tests_torch` shards.

  ## Who can review?

  @ArthurZucker, you mentioned in #31804 that you could help review this contribution. A review would be appreciated when you have time.

### CODE REVIEW

**Comment by ArthurZucker** [comment]
Hey! Happy to review, can you update the branch to fix the latest mlinter updates!

**Comment by sushmanthreddy** [comment]
@ArthurZucker sure give me 5 hours time , am in middle of my office work, will update and let you knew once ...there is little bit vibe coding have done on written tests ..I feel that code doesn't align with huggingface ..I will review myself code and ping over here again

**Comment by noanabeshima** [comment]
Added Apache-2.0 license to TinyModel @sushmanthreddy 

I'm probably happy to change it to whatever license is easiest in case you'd prefer another.

File: src/transformers/models/tiny_model/modeling_tiny_model.py
Comment (ArthurZucker) [inline src/transformers/models/tiny_model/modeling_tiny_model.py:1]: please create a module file for inheritance!

Comment (sushmanthreddy) [reply src/transformers/models/tiny_model/modeling_tiny_model.py:1]: sure @ArthurZucker ,I have other open PR also example ovis 2.5 is under review ,once that got merged ,I will come onto this pr .Iam just getting confused between models about architecture .

pls give me sometime to get ovis 2.5 get stable  state. I feel its almost over will pick this next
